python/ray/serve/pipeline/tests/test_pipeline.py

```diff
@@ -97,6 +96,9 @@ def test_class_constructor_not_called_until_deployed(execution_mode,
 
     with tempfile.NamedTemporaryFile("w") as tmp:
 
+        tmp_name = tmp.name + ".tmp"
+        open(tmp_name, "w+").close()
```

Comment:
Can we remove this if we just call `tempfile.NamedTemporaryFile("w+")` above?

Reply:
I don't think so. I have explained the reason in https://github.com/ray-project/ray/pull/21484#discussion_r780661542

`tempfile.NamedTemporaryFile("w+")` creates `C:\\CONDA_~1\\tmp04t9xjnp` which is without an extension `.tmp`. So, Windows confuses this with a directory leading to permission errors.

Reply:
I think as we discussed, we can use the `suffix` argument from https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile to fix this :)